test/scan/lexer_errors_public_spec.ts

```diff
@@ -0,0 +1,34 @@
+import { defaultLexerErrorProvider } from "../../src/scan/lexer_errors_public"
+import { IToken } from "../../api"
+
+describe("The Chevrotain default lexer error message provider", () => {
```

Comment:
These unit tests are good, Although they are already covered in more complex tests and lexer_spec.
But not exactly what I meant when I said tests need to be added.
- But keep them, it is always good to have more unit tests (especially when they run quickly).

- So we have tests for the building of the error messages string.
- And we have existing **integration** tests that show the default error message provider is working correctly in a **full flow** when running the lexer.

- But where are the tests that show a **custom** error message provider works? 😄